test/Core.Test/Utilities/CoreHelpersTests.cs

```diff
@@ -0,0 +1,221 @@
+            // the comb are working properly
+        }
+
+        [Theory]
+        [InlineData(2, 5, new[] { 1, 2, 3, 4, 5, 6, 7, 8 , 9, 0 })]
+        [InlineData(2, 3, new[] { 1, 2, 3, 4, 5 })]
+        [InlineData(2, 1, new[] { 1, 2 })]
+        [InlineData(1, 1, new[] { 1 })]
+        [InlineData(2, 2, new[] { 1, 2, 3 })]
+        public void Batch_Success(int batchSize, int totalBatches, int[] collection)
+        {
+            // Arrange
+            var remainder = collection.Length % batchSize;
+
+            // Act
+            var batches = collection.Batch(batchSize);
+
+            // Assert
+            Assert.Equal(totalBatches, batches.Count());
+
+            foreach (var batch in batches.Take(totalBatches - 1))
+            {
+                Assert.Equal(batchSize, batch.Count());
+            }
+
+            Assert.Equal(batches.Last().Count(), remainder == 0 ? batchSize : remainder);
+        }
+
+        [Fact]
+        public void ToGuidIdArrayTVP_Success()
+        {
+            // Arrange
+            var item0 = Guid.NewGuid();
+            var item1 = Guid.NewGuid();
+
+            var ids = new[] { item0, item1 };
+
+            // Act
+            var dt = ids.ToGuidIdArrayTVP();
+
+            // Assert
+            Assert.Single(dt.Columns);
+            Assert.Equal("GuidId", dt.Columns[0].ColumnName);
+            Assert.Equal(2, dt.Rows.Count);
+            Assert.Equal(item0, dt.Rows[0][0]);
+            Assert.Equal(item1, dt.Rows[1][0]);
+        }
+
+        // TODO: Test the other ToArrayTVP Methods
+
+        [Theory]
+        [InlineData("12345&6789", "123456789")]
+        [InlineData("abcdef", "ABCDEF")]
+        [InlineData("1!@#$%&*()_+", "1")]
+        [InlineData("\u00C6123abc\u00C7", "123ABC")]
+        [InlineData("123\u00C6ABC", "123ABC")]
+        [InlineData("\r\nHello", "E")]
+        [InlineData("\tdef", "DEF")]
+        [InlineData("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUV1234567890", "ABCDEFABCDEF1234567890")]
+        public void CleanCertificateThumbprint_Success(string input, string output)
+        {
+            // Arrange & Act
+            var sanitizedInput = CoreHelpers.CleanCertificateThumbprint(input);
+
+            // Assert
+            Assert.Equal(output, sanitizedInput);
+        }
+
+        // TODO: Add more tests
+        [Theory]
+        [InlineData("2020-12-30T11:49:12.0000000Z", 1609310952000L)]
+        public void ToEpocMilliseconds_Success(string dateString, long milliseconds)
+        {
+            // Arrange
+            var date = DateTime.Parse(dateString);
+
+            // Act & Assert
+            Assert.Equal(milliseconds, CoreHelpers.ToEpocMilliseconds(date));
+        }
+
+        [Theory]
+        [InlineData(1609310952000L, "2020-12-30T11:49:12.0000000Z")]
+        public void FromEpocMilliseconds(long milliseconds, string dateString)
+        {
+            // Arrange
+            var date = DateTime.Parse(dateString);
```

Comment:
I'm sorry about this but I'd assumed out pipelines were running on this PR and they aren't.

This doesn't pass for me locally. It seems DateTime.Parse is creating a local kind datetime despite the 'Z' in your string. ugh, I hate dates AND times.

The solution that worked for me locally was to use the [MemberData] provider and instantiate the DateTime as `new DateTime(2020,12,30,11,49,12, DateTimeKind.Utc)`. Then I think we can be sure it works regardless of local.

```
public static IEnumerable<object[]> _epochTestCases = new[] {
    new object[] {new DateTime(2020, 12, 30, 11, 49, 12, DateTimeKind.Utc), 1609328952000L},
};

// TODO: Add more tests
[Theory]
[MemberData(nameof(_epochTestCases))]
public void ToEpocMilliseconds_Success(DateTime date, long epochMilliseconds)
{
    // Act & Assert
    Assert.Equal(epochMilliseconds, CoreHelpers.ToEpocMilliseconds(date));
}

[Theory]
[MemberData(nameof(_epochTestCases))]
public void FromEpocMilliseconds(DateTime date, long epochMilliseconds)
{
    // Act & Assert
    Assert.Equal(date, CoreHelpers.FromEpocMilliseconds(epochMilliseconds));
}
```

Reply:
Out of curiosity I was wondering what it would take to get a Utc date from a string and this is what I came up with.  I still like your solution much better so I have implemented it but figured I would show you. 

```
var date = DateTime.ParseExact("2020-12-30T11:49:12.0000Z", 
    "yyyy-MM-ddThh:mm:ss.ffffZ", 
    CultureInfo.InvariantCulture, 
    DateTimeStyles.AdjustToUniversal);
```